### PR TITLE
Overhaul Point Generation 

### PR DESCRIPTION
**Pull request by TShapinsky**
- Bi-Directional points
- Ruby library for creating points for alfalfa via `EnergyPlus_Mixin` and `OpenStudio_Mixin` classes
- Better Reporting of certain E+ errors
- Convert `export_bcvtb` to use new library. This allows previous point generation constructs to still function.
- Pointed to `alfalfa_client` `develop` branch (this should fix the flakiness we were experiencing in CI) 

Added Points:

- Site Meters: (Electricity, Natural Gas)
- Zone Sensors: (Temp, Temp setpoint, Humidity)
- Schedules: (Heating/Cooling)

### CODE REVIEW

File: alfalfa_worker/jobs/openstudio/create_run.py [hunk 1]
```diff
@@ -43,7 +44,12 @@ def exec(self):
         check_output(['openstudio', str(lib_dir / 'merge_osws.rb'), str(default_workflow_path), str(submitted_osw_path)])
 
         # run workflow
-        check_output(['openstudio', 'run', '-m', '-w', str(submitted_osw_path)])
+        check_output(['openstudio', '--include', '/alfalfa/alfalfa_worker/jobs/openstudio/lib/alfalfa-lib/', 'run', '-m', '-w', str(submitted_osw_path)])
```
Comment (nllong) [inline alfalfa_worker/jobs/openstudio/create_run.py:47]: to handle the new OpenStudio Alfalfa Mixin?

File: alfalfa_worker/jobs/openstudio/lib/variables.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                result.update(json.load(fp))
+
+        return result
+
+    def load_reports(self):
+        self.inputs = self._load_json('**/*_report_inputs.json')
+        self.outputs = self._load_json('**/*_report_outputs.json')
+
+    def generate_variables_cfg(self) -> str:
+        bcvtb_variables_element = ElementTree.fromstring(
```
Comment (nllong) [inline alfalfa_worker/jobs/openstudio/lib/variables.py:38]: crazy but looks good. If we are going to handle XML a lot, then we should wrap the generation of these into an XML Class of some sort so we can validate.

File: alfalfa_worker/jobs/openstudio/lib/workflow/measures/export_bcvtb/measure.rb [hunk 3]
```diff
@@ -88,145 +64,124 @@ def run(model, runner, user_arguments)
...
-          prog.setBody(body)
-        end
-
-        emsOutputVariables.each do |outvar|
-          this_ems_var_name = outvar.emsVariableName
-          if this_ems_var_name == emsGlobalHandle
-            outvar.setEMSVariableName(eevarHandle)
-          end
-        end
+    ems_globals = model.getEnergyManagementSystemGlobalVariables
```
Comment (nllong) [inline alfalfa_worker/jobs/openstudio/lib/workflow/measures/export_bcvtb/measure.rb:93]: so much cleaner, thanks!

**Review by nllong — APPROVED** [review]
great updates. 

Code looks good, containers built, tests ran, integration tests ran, CI is happy!

Good work.



This is mostly OpenStudio related (except for the tweaking of models.py) , so just wondering how we will update the FMU side, if at all, to follow these patterns?


```
tests/integration/test_basic_operations.py::test_simple_internal_clock[model_path0] PASSED                                                             [  5%]
tests/integration/test_basic_operations.py::test_simple_internal_clock[model_path1] PASSED                                                             [ 11%]
tests/integration/test_basic_operations.py::test_simple_internal_clock[model_path2] PASSED                                                             [ 16%]
tests/integration/test_basic_operations.py::test_simple_internal_clock[model_path3] PASSED                                                             [ 22%]
tests/integration/test_basic_operations.py::test_simple_internal_clock[model_path4] PASSED                                                             [ 27%]
tests/integration/test_basic_operations.py::test_simple_external_clock[model_path0] PASSED                                                             [ 33%]
tests/integration/test_basic_operations.py::test_simple_external_clock[model_path1] PASSED                                                             [ 38%]
tests/integration/test_basic_operations.py::test_simple_external_clock[model_path2] PASSED                                                             [ 44%]
tests/integration/test_basic_operations.py::test_simple_external_clock[model_path3] PASSED                                                             [ 50%]
tests/integration/test_basic_operations.py::test_simple_external_clock[model_path4] PASSED                                                             [ 55%]
tests/integration/test_basic_operations.py::test_alias PASSED                                                                                          [ 61%]
tests/integration/test_broken_models.py::test_broken_models[broken_model_path0] PASSED                                                                 [ 66%]
tests/integration/test_refrig_case_osw.py::TestRefrigCaseOSW::test_basic_io PASSED                                                                     [ 72%]
tests/integration/test_refrig_case_osw.py::TestRefrigCaseOSW::test_invalid_start_conditions PASSED                                                     [ 77%]
tests/integration/test_simple_fmu_control.py::TestSimpleFMUControl::test_control SKIPPED (unconditional skip)                                          [ 83%]
tests/integration/test_simple_thermostat_fmu.py::TestSimpleThermostat::test_io_with_external_clock PASSED                                              [ 88%]
tests/integration/test_single_zone_vav_fmu.py::TestSingleZoneVAVFMU::test_simple_internal_clock PASSED                                                 [ 94%]
tests/integration/test_small_office_osw.py::test_python_environment PASSED                                                                             [100%]
```

**Review by axelstudios — APPROVED** [review]

**Pull request merged**